---
title: "How embryos acclimate to temperature through epigenetic regulation"
authors: 
  - "Thomas Oâ€™Leary, Emily Mikucki, Sumaetee Tangwancharoen, Sara Helms Cahan, Seth Frietze, Brent L. Lockwood"
format:
  html:
    theme: lumen
    fontsize: 95%
    toc: true
    toc-location: left
    number-sections: true
    self-contained: true
---

```{r setup}
#| include: false

# Set the default knitr options for the document
knitr::opts_chunk$set(
  echo = FALSE, 
  warning = FALSE, 
  message = FALSE
)

# Load all packages used in the analysis
library(Seurat)
library(Signac)
library(cowplot)
library(kableExtra)
library(tidyverse)
```

::: {.callout-note collapse="true"}
## To do list

#### Major {-}

- Figure sketch for paper
- Finish introduction
- Meet with Sara, Seth, Brent

#### Minor {-}

- I am [regressing out cell cycle genes](https://satijalab.org/seurat/articles/cell_cycle_vignette.html) from downstream analysis. Should I include supplementary figures for this or is it standard enough not to include?
- Shiny app minimal implementation
- Plotting themes and script and file standardization
- Consolidate script files and outputs
- Driver script

:::

# **Code** {-}

::: {.panel-tabset}

## Phenotype

```{r pheno, filename = "00_pheno.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/00_pheno/00_pheno.R"
```

## Cell Ranger ARC

::: {.callout-note}
## Cell Ranger ARC pipeline

All pre-processing of the multiome sequencing data using 10X Genomics Cell Ranger ARC pipeline was executed on the VACC.
:::

```{r mkref, filename = "00_mkref.sh"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/02_cellranger-arc/00_mkref.sh"
```

```{r count, filename = "01_count.sh"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/02_cellranger-arc/01_count.sh"
```

```{r aggr, filename = "02_aggr.sh"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/02_cellranger-arc/02_aggr.sh"
```

## Seurat

```{r create, filename = "00_create_seurat_object.R"}
#| echo: true
#| eval: false
#| code-fold: true
#| file: "../src/03_seurat/00_create_seurat_object.R"
```

```{r qc-filt, filename = "01_quality_control_filtering.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/01_quality_control_filtering.R"
```

```{r init-clust, filename = "02_initial_cluster.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/02_initial_cluster.R"
```

```{r amulet, filename = "03_amulet.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/03_amulet.R"
```

```{r doublet-finder, filename = "04_doublet_finder.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/04_doublet_finder.R"
```

```{r rm-mult, filename = "05_rm_multiplets.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/05_rm_multiplets.R"
```

```{r qc-cells, filename = "06_qc_data_cells.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/06_qc_data_cells.R"
```

```{r clust, filename = "07_cluster.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/07_cluster.R"
```

```{r markers, filename = "08_cluster_markers.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/08_cluster_markers.R"
```

```{r annot, filename = "09_cluster_annotation.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/09_cluster_annotation.R"
```

```{r link, filename = "10_link_peaks.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/10_link_peaks.R"
```

```{r degs, filename = "11_degs.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/11_degs.R"
```

```{r dars, filename = "12_dars.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/03_seurat/12_dars.R"
```

## Figures

```{r pheno-fig, filename = "pheno.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/04_plots/pheno.R"
```

```{r qc-fig, filename = "qc_initial.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/04_plots/qc_initial.R"
```

```{r qc_cells-fig, filename = "qc_cells.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/04_plots/qc_cells.R"
```

```{r mult-fig, filename = "multiplets.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/04_plots/multiplets.R"
```

```{r clust-fig, filename = "cluster.R"}
#| eval: false
#| echo: true
#| code-fold: true
#| file: "../src/04_plots/cluster.R"
```

:::

::: {.callout-note}
The code used for analysis is located [on GitHub](https://github.com/tsoleary/heater)
:::


# **Figure sketch** {-}

::: {.callout-caution}
## Rough outline of figures for the manuscript

Question for the supplementary figures: 

- What am I missing? Is there are quality control metric that should be included but isn't?
- Should I make S2 figure panels more variable in style (i.e. not all violin boxplots)?
:::

::: {.panel-tabset}

## Fig. 1 {-}

```{r fig-1}
#| fig-cap: "(A) Place holder for embryonic acclimation design. (B) Boxplots of egg hatching success after acute heat shock of embryos acclimated to different temperatures. Each point represents the proportion hatched from separate trials of at least 50 eggs."

knitr::include_graphics(
  here::here("output/figs/final/fig_1.png")
)
```

## Fig. 2 {-}

- Unknown
- UMAP with cell-type annotations

## Fig. S1 {-}

```{r fig-s1}
#| fig-cap: "Sequencing quality control and barcode filtering. Knee-plots of (A) ATAC counts and (B) RNA countsversus the top hundred thousand barcodes in rank order from most counts to least. (C) Scatter plot of RNA counts and ATAC counts with non-cell barcodes colored in grey and cells that pass filter criteria colored in green. Dashed lines represent the lower-bound count threshold of 800 and 200 counts for ATAC and RNA respecitively. Bar plots representing per sample quality control metrics including (D) number of cells, total number of mapped reads in cells in each sample for (E) the ATAC  (F) RNA libraries."

knitr::include_graphics(
  here::here("output/figs/final/fig_s1.png")
)
```

## Fig. S2 {-}

```{r fig-s2}
#| fig-cap: "Detailed quality control metrics for nuclei from ATAC- and RNA-sequenced libraries. Violin plot with overlayed boxplot showing the total number of reads per cell from RNA- (A) and ATAC- (C) sequencing in each sample. (B) The number of expressed genes detected per cell. (D) The number of unique ATAC features detected per cele. (E) Mean transcription start site (TSS) enrichment score profile. RNA reads mapped (G) ribosomes or genes in the (F) mitochondrial genome. (H) Nucelosome signal for each cell. (I) The fraction of ATAC reads in peaks (FRiP)."

knitr::include_graphics(
  here::here("output/figs/final/fig_s2.png")
)
```

## Fig. S3 {-}

Clusters and cell-type annotations

- Cluster UMAP split
- Number of cells per cluster
- Number of cells per cell-type annotation

:::

# **Overview**

## Experimental design

```{r acc-exp-design}
knitr::include_graphics(
  here::here("output/figs/exp_design/acc_design.png")
)
```

```{r hs-exp-design}
knitr::include_graphics(
  here::here("output/figs/exp_design/heatshock_design.png")
)
```

## Phenotype

```{r pheno-fig-print}
knitr::include_graphics(
  here::here("output/figs/pheno/cantonS_survival.png")
)
```

# **Quality control**

::: {.callout-caution collapse=true, icon=false}
## Criteria for filtering all barcodes to high-quality nuclei

1. We used Cell Ranger ARC called cell barcodes -- algorithm described [here](https://support.10xgenomics.com/single-cell-multiome-atac-gex/software/pipelines/latest/algorithms/cell-calling). 14,301 barcodes out of 2,215,183. 

2. Because the Cell Ranger ARC cell calling algorithm is very permissive to barcodes with very low counts (_i.e._, a minimum of a single count in each library), barcodes were additionally filtered to a low count threshold in both the ATAC and RNA libraries based on the clearly defined population of cells in the RNA and ATAC count scatterplot. Additionally, barcodes with more than 5% of RNA reads mapping to genes on the mitochondrial genome and/or more than 25% mapping to ribosomal genes were excluded.

3. Multiplets were filtered out using two independent methods, relying on either the ATAC or RNA libraries to call multiplets. [AMULET](https://github.com/UcarLab/AMULET) relies on the assumption that in snATAC-seq of diploids there should be at most two overlapping fragments with the same cell barcode. The presence of more than two overlapping fragments is a potential indication of a multiplet. Doublets were also identified using the RNA-seq libraries with [DoubletFinder](https://github.com/chris-mcginnis-ucsf/DoubletFinder). 
:::


## Knee plot

```{r knee}
#| label: fig-plot
#| fig-cap: "Knee plot of the number of ATAC reads and RNA reads in the raw data."

knitr::include_graphics(here::here("output/figs/qc/initial/knee_plots.png"))
```

## Scatter plot of counts

::: {.panel-tabset}

### Cells after all filtering

```{r all-scatter-fig}
knitr::include_graphics(
  here::here("output/figs/qc/cells/scatter_rna_atac_final_filter.png")
)
```

### Cell Ranger ARC only

```{r 10x-scatter-fig}
knitr::include_graphics(
  here::here("output/figs/qc/initial/scatter_rna_atac_10x_cells.png")
)
```

### And low count threshold

```{r low-scatter-fig}
knitr::include_graphics(
  here::here("output/figs/qc/initial/scatter_rna_atac_cells_threshold.png")
)
```

:::

## Violin QC plots

::: {.panel-tabset}

### After filtering

```{r qc-violin-post}
knitr::include_graphics(
  here::here("output/figs/qc/cells/violin_qc_metrics_post_filter.png")
)
```

### Before filtering

```{r qc-violin-pre}
knitr::include_graphics(
  here::here("output/figs/qc/initial/violin_qc_metrics.png")
)
```

:::

## Multiplets

::: {.panel-tabset}

### Mapped onto UMAP projection

```{r mult-umap}
knitr::include_graphics(
  here::here("output/figs/qc/multiplets/multiplet_umap.png")
)
```

### Counts

```{r counts-umap}
knitr::include_graphics(
  here::here("output/figs/qc/multiplets/counts_umap.png")
)
```

:::

### Count histograms {-}

::: {.panel-tabset}

#### Multiplets

```{r mult-hist}
knitr::include_graphics(
  here::here("output/figs/qc/multiplets/multiplets_hist.png")
)
```

#### After filtering

```{r sing-hist}
knitr::include_graphics(
  here::here("output/figs/qc/multiplets/singlets_only_hist.png")
)
```
:::

## QC metrics per sample

::: {.panel-tabset}

### Number of cells

```{r cells-samp}
knitr::include_graphics(
  here::here("output/figs/qc/cells/cells_sample.png")
)
```

### ATAC counts

```{r med-atac-samp}
knitr::include_graphics(
  here::here("output/figs/qc/cells/median_atac_sample.png")
)
```

### RNA counts

```{r med-rna-samp}
knitr::include_graphics(
  here::here("output/figs/qc/cells/median_rna_sample.png")
)
```

### TSS Enrichment

```{r med-tss-samp}
knitr::include_graphics(
  here::here("output/figs/qc/cells/tss_enrichment_samples.png")
)
```

### FRiP

```{r med-frip-samp}
knitr::include_graphics(
  here::here("output/figs/qc/cells/frip_samples.png")
)
```


### FRiT

```{r med-frit-samp}
knitr::include_graphics(
  here::here("output/figs/qc/cells/frit_samples.png")
)
```

:::

## QC metrics on UMAP

::: {.panel-tabset}

### ATAC counts

```{r umap-atac-count}
knitr::include_graphics(
  here::here("output/figs/qc/cells/features/nCount_ATAC.png")
)
```

### RNA counts

```{r umap-rna-count}
knitr::include_graphics(
  here::here("output/figs/qc/cells/features/nCount_RNA.png")
)
```

### TSS enrichment

```{r umap-tss}
knitr::include_graphics(
  here::here("output/figs/qc/cells/features/TSS.enrichment.png")
)
```

### Mito percent

```{r umap-mito}
knitr::include_graphics(
  here::here("output/figs/qc/cells/features/percent.mt.png")
)
```

### FRiP

```{r umap-frip}
knitr::include_graphics(
  here::here("output/figs/qc/cells/features/FRiP.png")
)
```

### FRiT

```{r umap-frit}
knitr::include_graphics(
  here::here("output/figs/qc/cells/features/FRiT.png")
)
```

:::


# **Clustering & annotation**

::: {.panel-tabset}

## Initial data processing

```{r clust-code}
#| eval: false
#| file: "../src/03_seurat/07_cluster.R"
```

## Notes

::: {.callout-note}
## Number of included dimensions

I kept the first 50 PCs. The elbow plot does not seem to entirely tail off before 50 PCs and the SCTransform method is a more robust normalization method that is less likely to carry artifacts do to technical variation. Please see [here](https://satijalab.org/seurat/archive/v3.0/sctransform_vignette.html) and [here](https://www.biostars.org/p/423306/).

The first LSI component is removed from downstream analysis because it is highly correlated with ATAC library depth. See below
:::

### PC elbow plot {-}

```{r elbow}
knitr::include_graphics(
  here::here("output/figs/cluster/qc/elbow_plot.png")
)
```

### LSI depth corr {-}

```{r lsi-depth}
knitr::include_graphics(
  here::here("output/figs/cluster/qc/lsi_depthcor.png")
)
```

:::

## Dimension reductions

::: {.panel-tabset}

### UMAP

::: {.panel-tabset}

### Joint {-}

```{r joint-umap}
knitr::include_graphics(
  here::here("output/figs/cluster/overlay/joint_umap.png")
)
```

### RNA-only {-}

```{r rna-umap}
knitr::include_graphics(
  here::here("output/figs/cluster/overlay/rna_umap.png")
)
```

### ATAC-only {-}

```{r atac-umap}
knitr::include_graphics(
  here::here("output/figs/cluster/overlay/atac_umap.png")
)
```

:::

### t-SNE

::: {.panel-tabset}

### Joint {-}

```{r joint-tsne}
knitr::include_graphics(
  here::here("output/figs/cluster/overlay/joint_tsne.png")
)
```

### RNA-only {-}

```{r rna-tsne}
knitr::include_graphics(
  here::here("output/figs/cluster/overlay/rna_tsne.png")
)
```

### ATAC-only {-}

```{r atac-tsne}
knitr::include_graphics(
  here::here("output/figs/cluster/overlay/atac_tsne.png")
)
```

:::

:::

## Clusters

::: {.callout-note}
## A note on clustering resolution

I have clustered the data at several resolutions from under-clustering to over-clustering. The approach that Calderon _et al._ took was to over-cluster, annotate cell-types for each cluster. Then merge back clusters based on matching cell-types. FOr example in the 4-6 hour development window. There were 31 seurat clusters, but only 11 annotated cell-types.

Here, I am presenting three clustering resolutions and their corresponding figures. 
:::

### Clustree

```{r clustree}
knitr::include_graphics(here::here("output/figs/cluster/qc/clustree.png"))
```

::: {.panel-tabset}

### `res = 0.1` {-}

#### Acclimation clusters {-}

::: {.panel-tabset}

##### Joint {-}

```{r res_0.1-umap}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/joint_umap.png")
)
```

##### RNA-only {-}

```{r res_0.1-umap_rna}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/rna_umap.png")
)
```

##### ATAC-only {-}

```{r res_0.1-umap_atac}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/atac_umap.png")
)
```

:::

#### Cells per cluster {-}

```{r res_0.1-n-cells-clust}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/cells_per_cluster.png")
)
```


#### Cluster specific quality control {-}

::: {.panel-tabset}

#### RNA count

```{r res_0.1-rna}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/rna_vln.png")
)
```

#### ATAC count

```{r res_0.1-atac}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/atac_vln.png")
)
```

#### Mito percent

```{r res_0.1-percent.mt}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/percent.mt_vln.png")
)
```

#### FRiP

```{r res_0.1-frip}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/frip_vln.png")
)
```

#### FRiT

```{r res_0.1-frit}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.1/frit_vln.png")
)
```

:::

### `res = 0.3` {-}

#### Acclimation clusters {-}

::: {.panel-tabset}

##### Joint {-}

```{r res_0.3-umap}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/joint_umap.png")
)
```

##### RNA-only {-}

```{r res_0.3-umap_rna}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/rna_umap.png")
)
```

##### ATAC-only {-}

```{r res_0.3-umap_atac}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/atac_umap.png")
)
```

:::

#### Cells per cluster {-}

```{r res_0.3-n-cells-clust}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/cells_per_cluster.png")
)
```


#### Cluster specific quality control {-}

::: {.panel-tabset}

#### RNA count

```{r res_0.3-rna}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/rna_vln.png")
)
```

#### ATAC count

```{r res_0.3-atac}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/atac_vln.png")
)
```

#### Mito percent

```{r res_0.3-percent.mt}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/percent.mt_vln.png")
)
```

#### FRiP

```{r res_0.3-frip}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/frip_vln.png")
)
```

#### FRiT

```{r res_0.3-frit}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.3/frit_vln.png")
)
```

:::

### `res = 0.7` {-}

#### Acclimation clusters {-}

::: {.panel-tabset}

##### Joint {-}

```{r res_0.7-umap}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/joint_umap.png")
)
```

##### RNA-only {-}

```{r res_0.7-umap_rna}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/rna_umap.png")
)
```

##### ATAC-only {-}

```{r res_0.7-umap_atac}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/atac_umap.png")
)
```

:::

#### Cells per cluster {-}

```{r res_0.7-n-cells-clust}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/cells_per_cluster.png")
)
```


#### Cluster specific quality control {-}

::: {.panel-tabset}

#### RNA count

```{r res_0.7-rna}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/rna_vln.png")
)
```

#### ATAC count

```{r res_0.7-atac}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/atac_vln.png")
)
```

#### Mito percent

```{r res_0.7-percent.mt}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/percent.mt_vln.png")
)
```

#### FRiP

```{r res_0.7-frip}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/frip_vln.png")
)
```

#### FRiT

```{r res_0.7-frit}
knitr::include_graphics(
  here::here("output/figs/cluster/res/res_0.7/frit_vln.png")
)
```

:::

:::

## Marker genes for `res = 0.7` clusters

```{r marker-genes}
readRDS(here::here("output/markers/cluster_markers.rds")) |> 
  mutate(cluster = as.numeric(cluster)) |> 
  group_by(cluster) |> 
  slice_min(minimump_p_val, n = 10, with_ties = FALSE) |> 
  kable(format = "html", escape = F) |> 
  kable_styling("striped") |> 
  scroll_box(width = "100%", height = "450px")
```

## Annotation

::: {.callout-note}
## Resolution

I have picked `res = 0.7` for the downstream analysis. 
:::


::: {.panel-tabset}

### UMAP

```{r}
knitr::include_graphics(
  here::here("output/figs/annot/umap_18_25_split.png")
)
```

### Cluster to cell-type

```{r annot-man}
readRDS(here::here("data/processed/annot/annot.rds")) |> 
  kable(format = "html", escape = F) |>
  kable_styling("striped") |>
  scroll_box(width = "100%", height = "450px")
```

### Top annotation results

```{r annot-res}
df <- readRDS(here::here("data/processed/annot/cluster_annot_all.rds")) |>
  mutate(cluster = as.numeric(cluster)) |>
  filter(padj < 0.05 | prop_overlap > 0.01) |>
  group_by(cluster) |>
  slice_min(padj, n = 5) |>
  arrange(cluster, padj)

df |>
  kable(format = "html", escape = F) |>
  kable_styling("striped") |>
  row_spec(which(df$padj < 0.05), color = "firebrick") |>
  scroll_box(width = "100%", height = "500px")
```

## Cell-type stats

::: {.panel-tabset}

### Number of genes

```{r}
knitr::include_graphics(
  here::here("output/figs/annot/genes_per_celltype.png")
)
```

### Number of peaks

```{r}
knitr::include_graphics(
  here::here("output/figs/annot/peaks_per_celltype.png")
)
```

:::

:::

# **Differential expression & accessibility**

### RNA

#### Cell-type

```{r deg-clust}
degs <- readRDS(here::here("output/degs/degs_cell-type.rds")) |> 
  filter(p_val_adj < 0.05)
```

::: {.panel-tabset}

### Number of DEGs

```{r n-degs-clust}
knitr::include_graphics(
  here::here("output/figs/degs/degs_count.png")
)
```

### Table of DEGs

```{r tab-degs-clust}
degs |> 
  kable(format = "html", escape = F) |> 
  kable_styling("striped") |> 
  scroll_box(width = "100%", height = "450px")
```

:::


## Psuedobulk

::: {.panel-tabset}

### Volcano plot

```{r vol-plot-}
knitr::include_graphics(
  here::here("output/figs/degs/pseudobulk_volcano.png")
)
```

### Table

```{r pseudo-degseq}
pseudobulk_DESeq <- readRDS(here::here("output/degs/pseudobulk_DESeq.rds")) |> 
  filter(padj < 0.05) 

pseudobulk_DESeq |> 
  kable(format = "html", escape = F) |> 
  kable_styling("striped") |> 
  scroll_box(width = "100%", height = "450px")
```



:::

## ATAC

### Psuedobulk

::: {.panel-tabset}

#### Volcano plot

```{r vol-plot-dars}
knitr::include_graphics(
  here::here("output/figs/dars/pseudobulk_volcano.png")
)
```

#### Table

```{r pseudo-dars}
pseudobulk_DESeq <- readRDS(here::here("output/dars/dars.rds"))

pseudobulk_DESeq |> 
  kable(format = "html", escape = F) |> 
  kable_styling("striped") |> 
  scroll_box(width = "100%", height = "450px")
```

:::

::: {.panel-tabset}

### Number of DARs per Cell-type {-}

```{r, n-dars}
knitr::include_graphics(
  here::here("output/figs/dars/dars_celltype.png")
)
```

### Table of DARs {-}

```{r dar-tab}
readRDS(here::here("output/dars/dars_cell-type_gene.rds")) |> 
  filter(p_val_adj < 0.05) |> 
  kable(format = "html", escape = F) |> 
  kable_styling("striped") |> 
  scroll_box(width = "100%", height = "450px")
```

:::

### A few examples

::: {.panel-tabset}

#### _Ldh_ {-}

```{r roi-ldh}
roi <- "3L-6262342-6264639.png"

knitr::include_graphics(
  here::here("output/figs/dars/coverage_plots", roi)
)
```

#### _lola_ {-}

```{r roi-lola}
roi <- "2R-10486272-10486985.png"

knitr::include_graphics(
  here::here("output/figs/dars/coverage_plots", roi)
)
```



#### _mbl_ {-}

```{r roi-mbl}
roi <- "2R-17261661-17263174.png"

knitr::include_graphics(
  here::here("output/figs/dars/coverage_plots", roi)
)
```

#### _lac_ {-}

```{r roi-lac}
roi <- "2R-12460368-12461108.png"

knitr::include_graphics(
  here::here("output/figs/dars/coverage_plots", roi)
)
```

#### _Ref2_ {-}

```{r roi-ref2}
roi <- "2L-12611799-12612226.png"

knitr::include_graphics(
  here::here("output/figs/dars/coverage_plots", roi)
)
```

:::

## DARs _near_ DEGs

### Table of DARs near DEGs

::: {.callout-note}
There are only two differentially expressed genes with nearby differentially accessible regions.

_Near_ is defined as a peak within 5,000 bp's of a gene.
:::

```{r dar-deg-tab}
dars_degs <- readRDS(here::here("output/dars/dars_degs.rds"))

dars_degs |> 
  select(-c("chrom", "start", "end")) |> 
  select(cell_type, region, gene, dist, everything()) |> 
  kable(format = "html", escape = F) |> 
  add_header_above(c(" " = 4, "ATAC" = 6, "RNA" = 6)) |> 
  kable_styling("striped") |> 
  scroll_box(width = "100%", height = "450px") 
```

#### Scatter of log-fold changes

::: {.callout-tip}
## Matching log-fold changes

Observed directionality of log-fold changes in expression and accessibility match.
:::

```{r dar-deg-scatter}
knitr::include_graphics(
  here::here("output/figs/dars/dars_degs_scatter.png")
)
```

### Accesibility and expression

::: {.panel-tabset}

#### _CG43759_ {-}

```{r goi-dar-deg-cg43759}
goi <- "CG43759.png"

knitr::include_graphics(
  here::here("output/figs/dars/cov_exp", goi)
)
```

#### _chn_ {-}

```{r goi-dar-deg-chn}
goi <- "chn.png"

knitr::include_graphics(
  here::here("output/figs/dars/cov_exp", goi)
)
```

:::

# **Weighted correlation network analysis**

## RNA

### Overview plots

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna", "softpower_threshold.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna", "dendrogram.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna", "module_correlogram.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna", "module_feature_plot.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna", "DME_lolipops.png")
)
```

### Significant modules

#### Red

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/red/red_all.png")
)
```

#### Brown

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/brown/brown_top_100.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/brown/brown_top_101-200.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/brown/brown_top_201-300.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/brown/brown_top_301-346.png")
)
```

## ATAC

### Overview plots

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/atac", "softpower_threshold.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/atac", "dendrogram.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/atac", "module_correlogram.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/atac", "module_feature_plot.png")
)
```

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/atac", "DME_lolipops.png")
)
```


### Significant modules

#### Turquoise

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/atac/turquoise_20.png")
)
```

#### Brown

```{r}
knitr::include_graphics(
  here::here("output/figs/wgcna/atac/brown_20.png")
)
```

# **SCENIC+**

Check out [this link](https://github.com/tsoleary/heater/blob/main/src/scenic_plus.ipynb) for the full `scenic_plus.ipynb` script and output!

```{r}
knitr::include_graphics(
  here::here("output/figs/scenicplus/eRegulon_dotheatmap.png")
)
```


# Session Info {-}

```{r session-info}
#| echo: false
#| eval: true
sessionInfo()
```