<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Cell Ranger ARC | How Embryos Acclimate to Temperature through Epigenetic Regulation</title>
  <meta name="description" content="Record of analysis for HEATER project." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Cell Ranger ARC | How Embryos Acclimate to Temperature through Epigenetic Regulation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Record of analysis for HEATER project." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Cell Ranger ARC | How Embryos Acclimate to Temperature through Epigenetic Regulation" />
  
  <meta name="twitter:description" content="Record of analysis for HEATER project." />
  

<meta name="author" content="Thomas Oâ€™Leary, Emily Mikucki, Sumaetee Tangwancharoen" />
<meta name="author" content="Sara Helms Cahan, Seth Frietze, and Brent L. Lockwood" />


<meta name="date" content="2023-05-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="nuclei-isolation.html"/>
<link rel="next" href="seurat.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="heater.html"><a href="heater.html"><i class="fa fa-check"></i><strong>HEATER</strong></a></li>
<li class="chapter" data-level="1" data-path="embryonic-thermal-acclimation.html"><a href="embryonic-thermal-acclimation.html"><i class="fa fa-check"></i><b>1</b> <strong>Embryonic thermal acclimation</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="embryonic-thermal-acclimation.html"><a href="embryonic-thermal-acclimation.html#egg-hatching-success"><i class="fa fa-check"></i><b>1.1</b> Egg hatching success</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="nuclei-isolation.html"><a href="nuclei-isolation.html"><i class="fa fa-check"></i><b>2</b> <strong>Nuclei isolation</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="nuclei-isolation.html"><a href="nuclei-isolation.html#nuclei-quality"><i class="fa fa-check"></i><b>2.1</b> Nuclei quality</a></li>
<li class="chapter" data-level="2.2" data-path="nuclei-isolation.html"><a href="nuclei-isolation.html#nuclei-counts"><i class="fa fa-check"></i><b>2.2</b> Nuclei counts</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cell-ranger-arc.html"><a href="cell-ranger-arc.html"><i class="fa fa-check"></i><b>3</b> <strong>Cell Ranger ARC</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="cell-ranger-arc.html"><a href="cell-ranger-arc.html#mkref"><i class="fa fa-check"></i><b>3.1</b> <code>mkref</code></a></li>
<li class="chapter" data-level="3.2" data-path="cell-ranger-arc.html"><a href="cell-ranger-arc.html#count"><i class="fa fa-check"></i><b>3.2</b> <code>count</code></a></li>
<li class="chapter" data-level="3.3" data-path="cell-ranger-arc.html"><a href="cell-ranger-arc.html#aggr"><i class="fa fa-check"></i><b>3.3</b> <code>aggr</code></a></li>
<li class="chapter" data-level="3.4" data-path="cell-ranger-arc.html"><a href="cell-ranger-arc.html#summary-of-output"><i class="fa fa-check"></i><b>3.4</b> Summary of output</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="seurat.html"><a href="seurat.html"><i class="fa fa-check"></i><b>4</b> <strong>Seurat</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="seurat.html"><a href="seurat.html#create-seurat-object"><i class="fa fa-check"></i><b>4.1</b> Create seurat object</a></li>
<li class="chapter" data-level="4.2" data-path="seurat.html"><a href="seurat.html#quality-control-and-filtering"><i class="fa fa-check"></i><b>4.2</b> Quality control and filtering</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i>Session Info</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">How Embryos Acclimate to Temperature through Epigenetic Regulation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cell-ranger-arc" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> <strong>Cell Ranger ARC</strong><a href="cell-ranger-arc.html#cell-ranger-arc" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="mkref" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> <code>mkref</code><a href="cell-ranger-arc.html#mkref" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="cell-ranger-arc.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb3-2"><a href="cell-ranger-arc.html#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="cell-ranger-arc.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a reference package for D. melanogaster ---------------------------------</span></span>
<span id="cb3-4"><a href="cell-ranger-arc.html#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="cell-ranger-arc.html#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Directions for running this script -------------------------------------------</span></span>
<span id="cb3-6"><a href="cell-ranger-arc.html#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 0. Be sure the following things have happened before beginning</span></span>
<span id="cb3-7"><a href="cell-ranger-arc.html#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   - Download FASTA and GTF files for D. melanogaster from Ensembl</span></span>
<span id="cb3-8"><a href="cell-ranger-arc.html#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   - Create a contig file according to 10X Genomics instructions</span></span>
<span id="cb3-9"><a href="cell-ranger-arc.html#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. In the command line, run the following command: bash path/to/this/file.sh</span></span>
<span id="cb3-10"><a href="cell-ranger-arc.html#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="cell-ranger-arc.html#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up directories and variables ---------------------------------------------</span></span>
<span id="cb3-12"><a href="cell-ranger-arc.html#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="cell-ranger-arc.html#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Move to the directory where you want the output files to be saved</span></span>
<span id="cb3-14"><a href="cell-ranger-arc.html#cb3-14" aria-hidden="true" tabindex="-1"></a>cd <span class="sc">/</span>netfiles02<span class="sc">/</span>lockwood_lab<span class="sc">/</span>heater<span class="sc">/</span>data<span class="sc">/</span>processed<span class="sc">/</span>seq<span class="sc">/</span>ref<span class="sc">/</span></span>
<span id="cb3-15"><a href="cell-ranger-arc.html#cb3-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-16"><a href="cell-ranger-arc.html#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Point towards installed cellranger-arc program</span></span>
<span id="cb3-17"><a href="cell-ranger-arc.html#cb3-17" aria-hidden="true" tabindex="-1"></a>export PATH<span class="ot">=</span><span class="er">/</span>netfiles02<span class="sc">/</span>lockwood_lab<span class="sc">/</span>cellranger<span class="sc">-</span>arc<span class="sc">/</span>cellranger<span class="sc">-</span>arc<span class="dv">-2</span>.<span class="fl">0.2</span><span class="sc">:</span><span class="er">$</span>PATH</span>
<span id="cb3-18"><a href="cell-ranger-arc.html#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="cell-ranger-arc.html#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="co"># # Set variables for the script</span></span>
<span id="cb3-20"><a href="cell-ranger-arc.html#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="co"># ref_path=&quot;/netfiles02/lockwood_lab/heater/data/processed/seq/ref/&quot;</span></span>
<span id="cb3-21"><a href="cell-ranger-arc.html#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="co"># libraries_path=&quot;/netfiles02/lockwood_lab/heater/data/raw/seq/libraries/&quot;</span></span>
<span id="cb3-22"><a href="cell-ranger-arc.html#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="cell-ranger-arc.html#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a raw unfiltered reference package ------------------------------------</span></span>
<span id="cb3-24"><a href="cell-ranger-arc.html#cb3-24" aria-hidden="true" tabindex="-1"></a>cellranger<span class="sc">-</span>arc mkref <span class="sc">--</span>config<span class="ot">=</span>BDGP6.<span class="fl">32.</span>config</span>
<span id="cb3-25"><a href="cell-ranger-arc.html#cb3-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-26"><a href="cell-ranger-arc.html#cb3-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the GTF to restrict it to protein-coding, lncRNA, antisense, and </span></span>
<span id="cb3-27"><a href="cell-ranger-arc.html#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   immune-related genes as 10X Genomics has done for their human and mouse refs</span></span>
<span id="cb3-28"><a href="cell-ranger-arc.html#cb3-28" aria-hidden="true" tabindex="-1"></a>cellranger<span class="sc">-</span>arc mkgtf Drosophila_melanogaster.BDGP6.<span class="dv">32</span>.<span class="fl">109.</span>gtf \</span>
<span id="cb3-29"><a href="cell-ranger-arc.html#cb3-29" aria-hidden="true" tabindex="-1"></a>  Drosophila_melanogaster.BDGP6.<span class="fl">32.109</span>_filtered.gtf \</span>
<span id="cb3-30"><a href="cell-ranger-arc.html#cb3-30" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>protein_coding \</span>
<span id="cb3-31"><a href="cell-ranger-arc.html#cb3-31" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>lncRNA \</span>
<span id="cb3-32"><a href="cell-ranger-arc.html#cb3-32" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>antisense \</span>
<span id="cb3-33"><a href="cell-ranger-arc.html#cb3-33" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>IG_LV_gene \</span>
<span id="cb3-34"><a href="cell-ranger-arc.html#cb3-34" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>IG_V_gene \</span>
<span id="cb3-35"><a href="cell-ranger-arc.html#cb3-35" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>IG_V_pseudogene \</span>
<span id="cb3-36"><a href="cell-ranger-arc.html#cb3-36" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>IG_D_gene \</span>
<span id="cb3-37"><a href="cell-ranger-arc.html#cb3-37" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>IG_J_gene \</span>
<span id="cb3-38"><a href="cell-ranger-arc.html#cb3-38" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>IG_J_pseudogene \</span>
<span id="cb3-39"><a href="cell-ranger-arc.html#cb3-39" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>IG_C_gene \</span>
<span id="cb3-40"><a href="cell-ranger-arc.html#cb3-40" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>IG_C_pseudogene \</span>
<span id="cb3-41"><a href="cell-ranger-arc.html#cb3-41" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>TR_V_gene \</span>
<span id="cb3-42"><a href="cell-ranger-arc.html#cb3-42" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>TR_V_pseudogene \</span>
<span id="cb3-43"><a href="cell-ranger-arc.html#cb3-43" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>TR_D_gene \</span>
<span id="cb3-44"><a href="cell-ranger-arc.html#cb3-44" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>TR_J_gene \</span>
<span id="cb3-45"><a href="cell-ranger-arc.html#cb3-45" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>TR_J_pseudogene \</span>
<span id="cb3-46"><a href="cell-ranger-arc.html#cb3-46" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>attribute<span class="ot">=</span>gene_biotype<span class="sc">:</span>TR_C_gene</span>
<span id="cb3-47"><a href="cell-ranger-arc.html#cb3-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-48"><a href="cell-ranger-arc.html#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a filtered reference package ------------------------------------------</span></span>
<span id="cb3-49"><a href="cell-ranger-arc.html#cb3-49" aria-hidden="true" tabindex="-1"></a>cellranger<span class="sc">-</span>arc mkref <span class="sc">--</span>config<span class="ot">=</span>BDGP6.<span class="fl">32.</span>filtered.config </span></code></pre></div>
</div>
<div id="count" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> <code>count</code><a href="cell-ranger-arc.html#count" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="cell-ranger-arc.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb4-2"><a href="cell-ranger-arc.html#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="cell-ranger-arc.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Directions for running this script -------------------------------------------</span></span>
<span id="cb4-4"><a href="cell-ranger-arc.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 0. Be sure the following things have happened before beginning</span></span>
<span id="cb4-5"><a href="cell-ranger-arc.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   - Demultiplex all samples</span></span>
<span id="cb4-6"><a href="cell-ranger-arc.html#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   - Create reference package</span></span>
<span id="cb4-7"><a href="cell-ranger-arc.html#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   - Create a separate libraries csv file pointing to demuxed fastq files</span></span>
<span id="cb4-8"><a href="cell-ranger-arc.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   - Set up slurm.template file according to 10X Genomics instructions in </span></span>
<span id="cb4-9"><a href="cell-ranger-arc.html#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#      cellranger-arc-2.0.2/external/martian/jobmanagers</span></span>
<span id="cb4-10"><a href="cell-ranger-arc.html#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. In the command line, run the following command: sbatch path/to/this/file.sh</span></span>
<span id="cb4-11"><a href="cell-ranger-arc.html#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="cell-ranger-arc.html#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up directories and variables ---------------------------------------------</span></span>
<span id="cb4-13"><a href="cell-ranger-arc.html#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="cell-ranger-arc.html#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Move to the directory where you want the output files to be saved</span></span>
<span id="cb4-15"><a href="cell-ranger-arc.html#cb4-15" aria-hidden="true" tabindex="-1"></a>cd <span class="sc">/</span>netfiles02<span class="sc">/</span>lockwood_lab<span class="sc">/</span>heater<span class="sc">/</span>data<span class="sc">/</span>processed<span class="sc">/</span>seq<span class="sc">/</span>samples<span class="sc">/</span></span>
<span id="cb4-16"><a href="cell-ranger-arc.html#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="cell-ranger-arc.html#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Point towards installed cellranger-arc program</span></span>
<span id="cb4-18"><a href="cell-ranger-arc.html#cb4-18" aria-hidden="true" tabindex="-1"></a>export PATH<span class="ot">=</span><span class="er">/</span>netfiles02<span class="sc">/</span>lockwood_lab<span class="sc">/</span>cellranger<span class="sc">-</span>arc<span class="sc">/</span>cellranger<span class="sc">-</span>arc<span class="dv">-2</span>.<span class="fl">0.2</span><span class="sc">:</span><span class="er">$</span>PATH</span>
<span id="cb4-19"><a href="cell-ranger-arc.html#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="cell-ranger-arc.html#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Set variables for the script</span></span>
<span id="cb4-21"><a href="cell-ranger-arc.html#cb4-21" aria-hidden="true" tabindex="-1"></a>ref_path<span class="ot">=</span><span class="st">&quot;/netfiles02/lockwood_lab/heater/data/raw/seq/ref/BDGP6_32_filtered&quot;</span></span>
<span id="cb4-22"><a href="cell-ranger-arc.html#cb4-22" aria-hidden="true" tabindex="-1"></a>libraries_path<span class="ot">=</span><span class="st">&quot;/netfiles02/lockwood_lab/heater/data/raw/seq/libraries/&quot;</span></span>
<span id="cb4-23"><a href="cell-ranger-arc.html#cb4-23" aria-hidden="true" tabindex="-1"></a>n_cores<span class="ot">=</span><span class="dv">32</span></span>
<span id="cb4-24"><a href="cell-ranger-arc.html#cb4-24" aria-hidden="true" tabindex="-1"></a>mem<span class="ot">=</span><span class="dv">64</span></span>
<span id="cb4-25"><a href="cell-ranger-arc.html#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="cell-ranger-arc.html#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Run cellranger-arc count for each sample -------------------------------------</span></span>
<span id="cb4-27"><a href="cell-ranger-arc.html#cb4-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-28"><a href="cell-ranger-arc.html#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample 18Â°C Rep 1</span></span>
<span id="cb4-29"><a href="cell-ranger-arc.html#cb4-29" aria-hidden="true" tabindex="-1"></a>cellranger<span class="sc">-</span>arc count <span class="sc">--</span>id<span class="ot">=</span>18C_Rep1 \</span>
<span id="cb4-30"><a href="cell-ranger-arc.html#cb4-30" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>reference<span class="ot">=</span><span class="er">$</span>{ref_path} \</span>
<span id="cb4-31"><a href="cell-ranger-arc.html#cb4-31" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>libraries<span class="ot">=</span><span class="er">$</span>{libraries_path}<span class="sc">/</span>18C_Rep1.csv \</span>
<span id="cb4-32"><a href="cell-ranger-arc.html#cb4-32" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>localcores<span class="ot">=</span><span class="er">$</span>{n_cores} \</span>
<span id="cb4-33"><a href="cell-ranger-arc.html#cb4-33" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>localmem<span class="ot">=</span><span class="er">$</span>{mem} \</span>
<span id="cb4-34"><a href="cell-ranger-arc.html#cb4-34" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>jobmode<span class="ot">=</span>slurm</span>
<span id="cb4-35"><a href="cell-ranger-arc.html#cb4-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-36"><a href="cell-ranger-arc.html#cb4-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample 18Â°C Rep 2</span></span>
<span id="cb4-37"><a href="cell-ranger-arc.html#cb4-37" aria-hidden="true" tabindex="-1"></a>cellranger<span class="sc">-</span>arc count <span class="sc">--</span>id<span class="ot">=</span>18C_Rep2 \</span>
<span id="cb4-38"><a href="cell-ranger-arc.html#cb4-38" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>reference<span class="ot">=</span><span class="er">$</span>{ref_path} \</span>
<span id="cb4-39"><a href="cell-ranger-arc.html#cb4-39" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>libraries<span class="ot">=</span><span class="er">$</span>{libraries_path}<span class="sc">/</span>18C_Rep2.csv \</span>
<span id="cb4-40"><a href="cell-ranger-arc.html#cb4-40" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>localcores<span class="ot">=</span><span class="er">$</span>{n_cores} \</span>
<span id="cb4-41"><a href="cell-ranger-arc.html#cb4-41" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>localmem<span class="ot">=</span><span class="er">$</span>{mem} \</span>
<span id="cb4-42"><a href="cell-ranger-arc.html#cb4-42" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>jobmode<span class="ot">=</span>slurm</span>
<span id="cb4-43"><a href="cell-ranger-arc.html#cb4-43" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb4-44"><a href="cell-ranger-arc.html#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample 25Â°C Rep 1</span></span>
<span id="cb4-45"><a href="cell-ranger-arc.html#cb4-45" aria-hidden="true" tabindex="-1"></a>cellranger<span class="sc">-</span>arc count <span class="sc">--</span>id<span class="ot">=</span>25C_Rep1 \</span>
<span id="cb4-46"><a href="cell-ranger-arc.html#cb4-46" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>reference<span class="ot">=</span><span class="er">$</span>{ref_path} \</span>
<span id="cb4-47"><a href="cell-ranger-arc.html#cb4-47" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>libraries<span class="ot">=</span><span class="er">$</span>{libraries_path}<span class="sc">/</span>25C_Rep1.csv \</span>
<span id="cb4-48"><a href="cell-ranger-arc.html#cb4-48" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>localcores<span class="ot">=</span><span class="er">$</span>{n_cores} \</span>
<span id="cb4-49"><a href="cell-ranger-arc.html#cb4-49" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>localmem<span class="ot">=</span><span class="er">$</span>{mem} \</span>
<span id="cb4-50"><a href="cell-ranger-arc.html#cb4-50" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>jobmode<span class="ot">=</span>slurm</span>
<span id="cb4-51"><a href="cell-ranger-arc.html#cb4-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-52"><a href="cell-ranger-arc.html#cb4-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample 25Â°C Rep 2</span></span>
<span id="cb4-53"><a href="cell-ranger-arc.html#cb4-53" aria-hidden="true" tabindex="-1"></a>cellranger<span class="sc">-</span>arc count <span class="sc">--</span>id<span class="ot">=</span>25C_Rep2 \</span>
<span id="cb4-54"><a href="cell-ranger-arc.html#cb4-54" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>reference<span class="ot">=</span><span class="er">$</span>{ref_path} \</span>
<span id="cb4-55"><a href="cell-ranger-arc.html#cb4-55" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>libraries<span class="ot">=</span><span class="er">$</span>{libraries_path}<span class="sc">/</span>25C_Rep2.csv \</span>
<span id="cb4-56"><a href="cell-ranger-arc.html#cb4-56" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>localcores<span class="ot">=</span><span class="er">$</span>{n_cores} \</span>
<span id="cb4-57"><a href="cell-ranger-arc.html#cb4-57" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>localmem<span class="ot">=</span><span class="er">$</span>{mem} \</span>
<span id="cb4-58"><a href="cell-ranger-arc.html#cb4-58" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>jobmode<span class="ot">=</span>slurm</span></code></pre></div>
</div>
<div id="aggr" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> <code>aggr</code><a href="cell-ranger-arc.html#aggr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="cell-ranger-arc.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb5-2"><a href="cell-ranger-arc.html#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="cell-ranger-arc.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate together all samples -----------------------------------------------</span></span>
<span id="cb5-4"><a href="cell-ranger-arc.html#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="cell-ranger-arc.html#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Directions for running this script -------------------------------------------</span></span>
<span id="cb5-6"><a href="cell-ranger-arc.html#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 0. Before running run cellranger-arc count for all samples and create </span></span>
<span id="cb5-7"><a href="cell-ranger-arc.html#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#      aggr_libraries.csv file pointing to atac_fragments.tsv.gz and </span></span>
<span id="cb5-8"><a href="cell-ranger-arc.html#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#      gex_molecule_info.h5 files per 10X Genomics instructions.</span></span>
<span id="cb5-9"><a href="cell-ranger-arc.html#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. In the command line, run the following command: sbatch path/to/this/file.sh</span></span>
<span id="cb5-10"><a href="cell-ranger-arc.html#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="cell-ranger-arc.html#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Request cluster resources ----------------------------------------------------</span></span>
<span id="cb5-12"><a href="cell-ranger-arc.html#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="cell-ranger-arc.html#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify partition</span></span>
<span id="cb5-14"><a href="cell-ranger-arc.html#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --partition=bluemoon</span></span>
<span id="cb5-15"><a href="cell-ranger-arc.html#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="cell-ranger-arc.html#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Request nodes</span></span>
<span id="cb5-17"><a href="cell-ranger-arc.html#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb5-18"><a href="cell-ranger-arc.html#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --ntasks-per-node=1</span></span>
<span id="cb5-19"><a href="cell-ranger-arc.html#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --cpus-per-task=12</span></span>
<span id="cb5-20"><a href="cell-ranger-arc.html#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="cell-ranger-arc.html#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Request memory for the entire job -- you can request --mem OR --mem-per-cpu</span></span>
<span id="cb5-22"><a href="cell-ranger-arc.html#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mem=31000</span></span>
<span id="cb5-23"><a href="cell-ranger-arc.html#cb5-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-24"><a href="cell-ranger-arc.html#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Reserve walltime -- hh:mm:ss -- 30 hrs max below</span></span>
<span id="cb5-25"><a href="cell-ranger-arc.html#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --time=30:00:00</span></span>
<span id="cb5-26"><a href="cell-ranger-arc.html#cb5-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-27"><a href="cell-ranger-arc.html#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Name this job</span></span>
<span id="cb5-28"><a href="cell-ranger-arc.html#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --job-name=aggr_all</span></span>
<span id="cb5-29"><a href="cell-ranger-arc.html#cb5-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-30"><a href="cell-ranger-arc.html#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Name output of this job using %x=job-name and %j=job-id</span></span>
<span id="cb5-31"><a href="cell-ranger-arc.html#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --output=logs/%x_%j.out </span></span>
<span id="cb5-32"><a href="cell-ranger-arc.html#cb5-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-33"><a href="cell-ranger-arc.html#cb5-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Echo some useful and interesting information </span></span>
<span id="cb5-34"><a href="cell-ranger-arc.html#cb5-34" aria-hidden="true" tabindex="-1"></a>echo <span class="st">&quot;Starting sbatch script at: `date`&quot;</span></span>
<span id="cb5-35"><a href="cell-ranger-arc.html#cb5-35" aria-hidden="true" tabindex="-1"></a>echo <span class="st">&quot;  running host:    ${SLURMD_NODENAME}&quot;</span></span>
<span id="cb5-36"><a href="cell-ranger-arc.html#cb5-36" aria-hidden="true" tabindex="-1"></a>echo <span class="st">&quot;  assigned nodes:  ${SLURM_JOB_NODELIST}&quot;</span></span>
<span id="cb5-37"><a href="cell-ranger-arc.html#cb5-37" aria-hidden="true" tabindex="-1"></a>echo <span class="st">&quot;  partition used:  ${SLURM_JOB_PARTITION}&quot;</span></span>
<span id="cb5-38"><a href="cell-ranger-arc.html#cb5-38" aria-hidden="true" tabindex="-1"></a>echo <span class="st">&quot;  jobid:           ${SLURM_JOBID}&quot;</span></span>
<span id="cb5-39"><a href="cell-ranger-arc.html#cb5-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-40"><a href="cell-ranger-arc.html#cb5-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up directories and variables ---------------------------------------------</span></span>
<span id="cb5-41"><a href="cell-ranger-arc.html#cb5-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-42"><a href="cell-ranger-arc.html#cb5-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Move to the directory where files will be aggregated</span></span>
<span id="cb5-43"><a href="cell-ranger-arc.html#cb5-43" aria-hidden="true" tabindex="-1"></a>cd <span class="sc">/</span>netfiles02<span class="sc">/</span>lockwood_lab<span class="sc">/</span>heater<span class="sc">/</span>data<span class="sc">/</span>processed<span class="sc">/</span>seq<span class="sc">/</span></span>
<span id="cb5-44"><a href="cell-ranger-arc.html#cb5-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-45"><a href="cell-ranger-arc.html#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Point towards installed cellranger-arc program</span></span>
<span id="cb5-46"><a href="cell-ranger-arc.html#cb5-46" aria-hidden="true" tabindex="-1"></a>export PATH<span class="ot">=</span><span class="er">/</span>netfiles02<span class="sc">/</span>lockwood_lab<span class="sc">/</span>cellranger<span class="sc">-</span>arc<span class="sc">/</span>cellranger<span class="sc">-</span>arc<span class="dv">-2</span>.<span class="fl">0.2</span><span class="sc">:</span><span class="er">$</span>PATH</span>
<span id="cb5-47"><a href="cell-ranger-arc.html#cb5-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-48"><a href="cell-ranger-arc.html#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Set variables for the script</span></span>
<span id="cb5-49"><a href="cell-ranger-arc.html#cb5-49" aria-hidden="true" tabindex="-1"></a>ref_path<span class="ot">=</span><span class="st">&quot;/netfiles02/lockwood_lab/heater/data/raw/seq/ref/BDGP6_32_filtered&quot;</span></span>
<span id="cb5-50"><a href="cell-ranger-arc.html#cb5-50" aria-hidden="true" tabindex="-1"></a>libraries_path<span class="ot">=</span><span class="st">&quot;/netfiles02/lockwood_lab/heater/data/processed/seq/samples/aggr_libraries.csv&quot;</span></span>
<span id="cb5-51"><a href="cell-ranger-arc.html#cb5-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-52"><a href="cell-ranger-arc.html#cb5-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Run cellranger-arc aggr to aggregate all samples together --------------------</span></span>
<span id="cb5-53"><a href="cell-ranger-arc.html#cb5-53" aria-hidden="true" tabindex="-1"></a>cellranger<span class="sc">-</span>arc aggr <span class="sc">--</span>id<span class="ot">=</span>all \</span>
<span id="cb5-54"><a href="cell-ranger-arc.html#cb5-54" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>csv<span class="ot">=</span><span class="er">$</span>{libraries_path} \</span>
<span id="cb5-55"><a href="cell-ranger-arc.html#cb5-55" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>reference<span class="ot">=</span><span class="er">$</span>{ref_path} \</span>
<span id="cb5-56"><a href="cell-ranger-arc.html#cb5-56" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>normalize<span class="ot">=</span>depth</span></code></pre></div>
</div>
<div id="summary-of-output" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Summary of output<a href="cell-ranger-arc.html#summary-of-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="cell-ranger-arc.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary stats from Cell Ranger ARC</span></span>
<span id="cb6-2"><a href="cell-ranger-arc.html#cb6-2" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data/processed/seq/all/outs/summary.csv&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb6-3"><a href="cell-ranger-arc.html#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">|&gt;</span></span>
<span id="cb6-4"><a href="cell-ranger-arc.html#cb6-4" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable</span>() <span class="sc">|&gt;</span></span>
<span id="cb6-5"><a href="cell-ranger-arc.html#cb6-5" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(<span class="st">&quot;striped&quot;</span>)</span></code></pre></div>
<pre><code>## Rows: 1 Columns: 45
## â”€â”€ Column specification â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
## Delimiter: &quot;,&quot;
## chr  (3): Sample ID, Genome, Pipeline version
## dbl (42): Number of cells, Number of cells (18C_Rep1), Number of cells (18C_...
## 
## â„¹ Use `spec()` to retrieve the full column specification for this data.
## â„¹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<table class="table table-striped" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Sample ID
</td>
<td style="text-align:left;">
all
</td>
</tr>
<tr>
<td style="text-align:left;">
Genome
</td>
<td style="text-align:left;">
BDGP6_32_filtered
</td>
</tr>
<tr>
<td style="text-align:left;">
Pipeline version
</td>
<td style="text-align:left;">
cellranger-arc-2.0.2
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:left;">
14301
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of cells (18C_Rep1)
</td>
<td style="text-align:left;">
2024
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of cells (18C_Rep2)
</td>
<td style="text-align:left;">
5188
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of cells (25C_Rep1)
</td>
<td style="text-align:left;">
3312
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of cells (25C_Rep2)
</td>
<td style="text-align:left;">
3777
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Fraction of genome in peaks
</td>
<td style="text-align:left;">
0.0854
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Fraction of high-quality fragments overlapping peaks
</td>
<td style="text-align:left;">
0.5032
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Fraction of reads kept (18C_Rep1)
</td>
<td style="text-align:left;">
0.6236
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Fraction of reads kept (18C_Rep2)
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Fraction of reads kept (25C_Rep1)
</td>
<td style="text-align:left;">
0.7992
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Fraction of reads kept (25C_Rep2)
</td>
<td style="text-align:left;">
0.8177
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Fraction of transposition events in peaks in cells
</td>
<td style="text-align:left;">
0.4692
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Median high-quality fragments per cell
</td>
<td style="text-align:left;">
6002
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Number of peaks
</td>
<td style="text-align:left;">
13927
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Post-normalization high-quality read pairs
</td>
<td style="text-align:left;">
213569184
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Post-normalization high-quality reads pairs per cell
</td>
<td style="text-align:left;">
14933.86
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs
</td>
<td style="text-align:left;">
262483329
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs (18C_Rep1)
</td>
<td style="text-align:left;">
61554884
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs (18C_Rep2)
</td>
<td style="text-align:left;">
66444472
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs (25C_Rep1)
</td>
<td style="text-align:left;">
66581198
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs (25C_Rep2)
</td>
<td style="text-align:left;">
67902775
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs per cell (18C_Rep1)
</td>
<td style="text-align:left;">
30412.49
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs per cell (18C_Rep2)
</td>
<td style="text-align:left;">
12807.34
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs per cell (25C_Rep1)
</td>
<td style="text-align:left;">
20103.02
</td>
</tr>
<tr>
<td style="text-align:left;">
ATAC Pre-normalization high-quality read pairs per cell (25C_Rep2)
</td>
<td style="text-align:left;">
17977.97
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Fraction of reads kept (18C_Rep1)
</td>
<td style="text-align:left;">
0.4455
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Fraction of reads kept (18C_Rep2)
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Fraction of reads kept (25C_Rep1)
</td>
<td style="text-align:left;">
0.6894
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Fraction of reads kept (25C_Rep2)
</td>
<td style="text-align:left;">
0.782
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Median UMI counts per cell
</td>
<td style="text-align:left;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Median genes per cell
</td>
<td style="text-align:left;">
606
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Post-normalization high-quality reads
</td>
<td style="text-align:left;">
154789965
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Post-normalization high-quality reads per cell
</td>
<td style="text-align:left;">
10823.72
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads
</td>
<td style="text-align:left;">
214036138
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads (18C_Rep1)
</td>
<td style="text-align:left;">
56426086
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads (18C_Rep2)
</td>
<td style="text-align:left;">
51237055
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads (25C_Rep1)
</td>
<td style="text-align:left;">
51522744
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads (25C_Rep2)
</td>
<td style="text-align:left;">
54850253
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads per cell (18C_Rep1)
</td>
<td style="text-align:left;">
27878.5
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads per cell (18C_Rep2)
</td>
<td style="text-align:left;">
9876.071
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads per cell (25C_Rep1)
</td>
<td style="text-align:left;">
15556.38
</td>
</tr>
<tr>
<td style="text-align:left;">
GEX Pre-normalization high-quality reads per cell (25C_Rep2)
</td>
<td style="text-align:left;">
14522.17
</td>
</tr>
</tbody>
</table>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="nuclei-isolation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="seurat.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": false,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
