---
title: "How Embryos Acclimate to Temperature through Epigenetic Regulation"
subtitle: "Figures"
author: 
  - "Thomas Oâ€™Leary, Emily Mikucki, Sumaetee Tangwancharoen, Sara Helms Cahan, Seth Frietze, and Brent L. Lockwood"
output:
  rmarkdown::html_document:
    theme: lumen
    number_sections: true
    toc: true
    toc_depth: 2
    toc_float: true
---

```{r, include = FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
library(tidyverse)
library(kableExtra)
library(bslib)


# Boxes
knitr::knit_hooks$set(
   error = function(x, options) {
     paste('\n\n<div class="alert alert-danger">',
           gsub('##', '\n', gsub('^##\ Error', '**Error**', x)),
           '</div>', sep = '\n')
   },
   warning = function(x, options) {
     paste('\n\n<div class="alert alert-warning">',
           gsub('##', '\n', gsub('^##\ Warning:', '**Warning**', x)),
           '</div>', sep = '\n')
   },
   message = function(x, options) {
     paste('\n\n<div class="alert alert-info">',
           gsub('##', '\n', x),
           '</div>', sep = '\n')
   }
)
```

# **Scripts**

The code used for analysis is located [on GitHub](https://github.com/tsoleary/heater) and compiled together [here](https://tsoleary.github.io/heater/docs/scripts.html).

# **Quality control**

## Calling nuclei barcodes

- First, we used Cell Ranger ARC called cell barcodes -- algorithm described [here](https://support.10xgenomics.com/single-cell-multiome-atac-gex/software/pipelines/latest/algorithms/cell-calling). 14,301 barcodes out of 2,215,183.
- Second, because the Cell Ranger ARC cell calling algorithm is very permissive to barcodes with very low counts (_i.e._, a minimum of a single count in each library), barcodes were additionally filtered to a low count threshold in both the ATAC and RNA libraries based on the clearly defined population of cells in the RNA and ATAC count scatterplot. Additionally, barcodes with more than 5% of RNA reads mapping to genes on the mitochondrial genome were excluded. A total of 886 barcodes were filtered out for 13,145 left.
- Last, multiplets were filtered out using two independent methods, relying on either the ATAC or RNA libraries to call multiplets. [AMULET](https://github.com/UcarLab/AMULET) relies on the assumption that in snATAC-seq of diploids there should be at most two overlapping fragments with the same cell barcode. The presence of more than two overlapping fragments is a potential indication of a multiplet. Doublets were also identified using the RNA-seq libraries with [DoubletFinder](https://github.com/chris-mcginnis-ucsf/DoubletFinder). After removing the 1,103 multiplet barcodes, there were still two barcodes that had substantially higher number of RNA reads after all filtering (i.e., n_Count_RNA around 12-13k when the next highest are around 6-7k). So I removed those two for a final number of 12,040 high quality nuceli barcodes.

## Knee plot

```{r}
knitr::include_graphics(here::here("output/figs/qc/knee_plots.png"))
```

## Scatter plot of counts {.tabset}

### Cells after all filtering {- .tabset}

```{r}
knitr::include_graphics(
  here::here("output/figs/qc/scatter_rna_atac_final_filter.png")
)
```

### Cell Ranger ARC only {- .tabset}

```{r}
knitr::include_graphics(
  here::here("output/figs/qc/scatter_rna_atac_10x_cells.png")
)
```

### And low count threshold {- .tabset}

```{r}
knitr::include_graphics(
  here::here("output/figs/qc/scatter_rna_atac_cells_threshold.png")
)
```

## Violin QC plots {.tabset}

### After filtering {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/violin_qc_metrics_post_filter.png"))
```

### Before filtering {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/violin_qc_metrics.png"))
```

## Multiplets {.tabset}

### Mapped onto UMAP projection  {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/multiplet_umap.png"))
```

### Counts {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/counts_umap.png"))
```

### Count histograms {-}

#### Multiplets {-}

```{r}
knitr::include_graphics(here::here("output/figs/qc/multiplets_hist.png"))
```

#### After filtering {-}

```{r}
knitr::include_graphics(here::here("output/figs/qc/singlets_only_hist.png"))
```

## QC metrics per sample  {.tabset}

### Number of cells {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/cells_sample.png"))
```

### ATAC counts {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/median_atac_sample.png"))
```

### RNA counts {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/median_rna_sample.png"))
```

### TSS Enrichment {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/tss_enrichment_samples.png"))
```

### FRiP {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/frip_samples.png"))
```


### FRiT {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/qc/frit_samples.png"))
```

## QC metrics on UMAP {.tabset}

### ATAC counts {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/features/qc/nCount_ATAC.png"))
```

### RNA counts {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/features/qc/nCount_RNA.png"))
```

### TSS enrichment {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/features/qc/TSS.enrichment.png"))
```

### Mito percent {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/features/qc/percent.mt.png"))
```

### FRiP {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/features/qc/FRiP.png"))
```

### FRiT {- .tabset}

```{r}
knitr::include_graphics(here::here("output/figs/features/qc/FRiT.png"))
```



# **Seurat**

## Clusters

### Cells per cluster {-}

```{r}
knitr::include_graphics(here::here("output/figs/cluster/cells_per_cluster.png"))
```

### Acclimation clusters {- .tabset}

#### Joint UMAP  {- .tabset .tabset-pills}

##### Split {-}

```{r}
knitr::include_graphics(
  here::here("output/figs/cluster/umap_18_25_split.png")
)
```

##### Together {-}

```{r, out.width = "75%"}
knitr::include_graphics(
  here::here("output/figs/cluster/umap_18_25_overlay.png")
)
```

#### RNA-only UMAP {- .tabset .tabset-pills}

##### Split {-}

```{r}
knitr::include_graphics(
  here::here("output/figs/cluster/umap_18_25_split_rna.png")
)
```

##### Together {-}

```{r, out.width = "75%"}
knitr::include_graphics(
  here::here("output/figs/cluster/umap_18_25_overlay_rna.png")
)
```

#### ATAC-only UMAP {- .tabset .tabset-pills}

##### Split {-}

```{r}
knitr::include_graphics(
  here::here("output/figs/cluster/umap_18_25_split_atac.png")
)
```

##### Together {-}

```{r, out.width = "75%"}
knitr::include_graphics(
  here::here("output/figs/cluster/umap_18_25_overlay_atac.png")
)
```

## Annotation

### Annotation results

```{r}
readRDS(here::here("data/processed/annot/cluster_annot_all.rds")) |> 
  mutate(cluster = as.numeric(cluster)) |> 
  filter(padj < 0.05) |> 
  arrange(cluster, padj) |> 
  kable(format = "html", escape = F) |> 
  kable_styling("striped") |> 
  scroll_box(width = "100%", height = "500px")
```

### Manual annotations

```{r}
readRDS(here::here("data/processed/annot/cluster_annot_manual.rds")) |> 
  kable(format = "html", escape = F) |> 
  kable_styling("striped") |> 
  scroll_box(width = "100%", height = "450px")
```

<div class="alert alert-info">
  <i class="fas fa-info-circle"></i> <strong>Note:</strong> Cluster 11 is likely also unknown or ubiquitous, as it does not have an annotation that passes FDR..
</div>


::: {.callout-note}
Note that there are five types of callouts, including:
`note`, `warning`, `important`, `tip`, and `caution`.
:::